File: Cpp/oneBillion/oneBillion.cpp
#include <cstdio>
#include <iostream>
#include <chrono>
#include <vector>

void counter(int *a, int b)
{
    while (*a != b) {
        if (*a < b) {
            (*a)++;
        } else if (*a > b) {
            (*a)--;
        }
    }
}

int main()
{
    int x = 1;
    int target = 1'000'000'000;

    auto start = std::chrono::high_resolution_clock::now();
    {
        counter(&x, target);
    }
    auto end = std::chrono::high_resolution_clock::now(); 

    auto duration = std::chrono::duration_cast<std::chrono::nanoseconds>(end - start).count();

    std::cout << x << std::endl;
    std::cout << "Time taken: " << duration << " nanoseconds" << std::endl;
    std::vector<int> blah;
    blah.insert();
    return 0;
}

File: Rust/test/oneBillion.rs
use std::time::Instant;

fn main() {
    let x: u64;
    let target: u64 = 1_000_000_000;

    let start = Instant::now(); // Start timing
    {
        x = (0..target).fold(0, |sum, _| sum+1);
    }
    let duration = start.elapsed(); // Calculate elapsed time

    println!("{}", x); // Print the final value of x
    println!("Time taken: {:?}", duration); // Print the duration
}

